src/Microsoft.Fx.Portability/DataExtensions.cs

```diff
@@ -33,13 +33,13 @@ public static class DataExtensions
 
         public static byte[] Serialize<T>(this T data)
         {
-            var str = JsonConvert.SerializeObject(data, Formatting.Indented, JsonSettings);
-
             using (var outputStream = new MemoryStream())
-            using (var writer = new StreamWriter(outputStream))
             {
-                writer.Write(str);
-                writer.Flush();
+                using (var writer = new StreamWriter(outputStream))
```

Comment:
The analyze request also gets compressed. We should rework these helpers so that it keeps items as streams as long as possible. The `.ToArray` creates an intermediary array that may not be needed and would just be excess allocations.